lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -3262,6 +3294,20 @@ swift::ModuleDecl *SwiftASTContext::GetModule(const SourceModule &module,
   // Create a diagnostic consumer for the diagnostics produced by the import.
   auto import_diags = getScopedDiagnosticConsumer();
 
+  // Report progress on module importing by using a callback function in
+  // swift::ASTContext
+  Progress progress("Importing Swift modules");
+  ast->SetPreModuleImportCallback([&progress](llvm::StringRef module_name,
+                                              bool is_overlay) {
+    progress.Increment(
+        1, (is_overlay ? module_name.str() + " (overlay)" : module_name.str()));
+  });
+
+  // Clear the callback function on scope exit to prevent an out-of-scope access
+  // of the progress local variable
+  auto on_exit = llvm::make_scope_exit(
+      [&]() { ast->SetPreModuleImportCallback(ReportModuleLoadingProgress); });
```

Comment:
Can you try setting the callback to an empty lambda instead of reusing `ReportModuleLoadingProgress` ?

Reply:
I'm using a generic callback here to report on any module imports progress reports not caught by the current callback lambdas but this can be replaced with a generic module import progress report.